community/csv/src/main/java/org/neo4j/csv/reader/SectionedCharBuffer.java

```diff
@@ -0,0 +1,196 @@
+/**
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.csv.reader;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.nio.CharBuffer;
+
+import static java.lang.Math.min;
+
+/**
+ * Has a similar role to a {@link CharBuffer}, but is tailored to how {@link BufferedCharSeeker}
+ * works and to be able to take full advantage of {@link ThreadAheadReadable}.
+ *
+ * First of all this thing wraps a {@code char[]} where the array, while still being one array,
+ * is sectioned into two equally sized parts: the back and the front. The flow of things is as follows:
+ * <ol>
+ * <li>Characters are read from a {@link Reader} using {@link #readFrom(Reader)} into the front section,
+ * i.e. starting from the middle of the array and forwards.</li>
```

Comment:
Well, it made sense when figuring out the terminology. Open to suggestions to make things more clear.